server/src/recipe/models/recipe-ingredient.ts

```diff
@@ -0,0 +1,53 @@
+import { IsInt, IsNotEmpty} from "class-validator";
+import { Types } from "mongoose";
+
+import { Validatable, ValidationGroup } from "../../app/index";
+import { FoodItemTemplate } from "./../../food/index";
+
+export class RecipeIngredient implements Validatable {
+
+  public _id: Types.ObjectId | string;
+  @IsNotEmpty({
+    groups: [ ValidationGroup.RECIPE ]
+  })
+  @IsInt({
+    groups: [ ValidationGroup.RECIPE ]
+  })
+  public amount: number;
+  @IsNotEmpty({
+    groups: [ ValidationGroup.RECIPE ]
+  })
+  public fooditemTemplateId: string | Types.ObjectId;
+  public fooditemTemplate: FoodItemTemplate;
+
+  public clone (recipeIngredient: RecipeIngredient) {
+    if (!recipeIngredient) {
+      return undefined;
+    }
+    this._id = recipeIngredient._id;
+    this.amount = recipeIngredient.amount;
+    this.fooditemTemplateId = recipeIngredient.fooditemTemplateId;
+    this.fooditemTemplate = new FoodItemTemplate().clone(recipeIngredient.fooditemTemplate);
```

Comment:
why have you the templateId and the template object?

Reply:
If you use the resolve query param the response contains the the whole fooditemTemplate else it only contains the id

Reply:
Okay. Should we mark the object as optional?

Reply:
Oh, I never thought about using the object in the request => If you specify a foodItemTemplate object in the request this would be ignored (since it's not part of the schema and so on)

But yes we can mark it as optional.